### PR TITLE
Create paginated device query

### PR DESCRIPTION
**Pull request by slevertbiot**

### CODE REVIEW

File: udmd/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+The project uses an Angular Front End and a GraphQL API Service to get data and present it.
+
+---
+
+### Running the Project
+
+1.  In order to be able to run both sub projects in parallel, you will need to install npm-run-all
+    ```
+    npm install -g npm-run-all
```
Comment (mspratt-biot) [inline udmd/README.md:11]: Neat

File: udmd/api/src/device/DeviceDataSource.ts [hunk 2]
```diff
@@ -37,7 +36,7 @@ export class DeviceDataSource extends GraphQLDataSource {
       const model: string = `model-${n}`;
       const site: string = `site-${n}`;
       const section: string = `section-${n}`;
-      const lastPayload: string = moment().format('YYYY-MM-DD');
+      const lastPayload: string = '2022-08-30';
```
Comment (mspratt-biot) [inline udmd/api/src/device/DeviceDataSource.ts:39]: FYI the lastPayload will be a ISO8601 utc timestamp. I know this is for examples sake, but just wanted to mention its what Trevor said.

File: udmd/package.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "udmd",
+  "version": "1.0.0",
+  "description": "simple package allowing us to run both sub projects with a single command",
+  "scripts": {
+    "web-build": "npm --prefix web/ install",
+    "web": "npm --prefix web/ run start",
```
Comment (mspratt-biot) [inline udmd/package.json:7]: Don't think you need the `run start`, can just be `start`, either way works.

File: udmd/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+/node_modules
```
Comment (mspratt-biot) [inline udmd/.gitignore:1]: So the node_modules outputs here instead of inside both api/ and web/ ?

Comment (slevertbiot) [reply udmd/.gitignore:1]: No, I had mistakenly installed npm-run-dev as a dev dependency and not global.  It ended up creating a node_modules folder.  I decided to be safe and exclude it anyway.  We should not see any node_module folder going forward unless we add dependencies.

File: udmd/README.md (hunk 1 above)
Comment (slevertbiot) [reply udmd/README.md:11]: say goodbye to docker-compose.

File: udmd/.gitignore (hunk 4 above)
Comment (mspratt-biot) [reply udmd/.gitignore:1]: Can we just remove it from the gitignore then? I'm surprised you also need to add .vscode. I thought I added .vscode and .DS_Store at the root gitignore allowing us to not need a gitignore in this udmd folder

**Review by mspratt-biot — APPROVED** [review]

File: udmd/.gitignore (hunk 4 above)
Comment (slevertbiot) [reply udmd/.gitignore:1]: Let me try removing the .gitignore.

**Review by mspratt-biot — APPROVED** [review]

**Pull request merged**